File: Bit Manipulation/missing_number.py
'''
Difficulty: Easy
https://leetcode.com/problems/missing-number/

Given an array nums containing n distinct numbers in the range [0, n], return the only number in the range that is missing from the array.


'''


class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        result = len(nums)

        for i in range(len(nums)):
            result += (i - nums[i])

        return result

File: 00.Rust/missing_number.rs
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let mut xor: i32 = 0;
        let n = nums.len();
        for n in 0..=n{
            xor ^= n as i32;
        }
        for n in nums{
            xor ^= n;
        }
        xor
    }
}